sidebar reorganization

pdg744 opened the pull request:
This PR mostly just moves content around in order to make the conceptual flow clearer. 
Rather than looking at the diff in GitHub, I suggest reviewing by opening the deployment, imagining yourself as a new user, and trying it out. 

The high-level thinking around the conceptual flow for users is described in this diagram: 
<img width="1046" alt="image" src="https://github.com/risc0/risc0/assets/15272444/08f89eee-adc6-48da-acf2-b905475f9ce2">

- [ ] This PR moves the `bonsai/quickstart.md` doc to `generating-proofs/remote-proving.md`. If we use this organization, we need to set up a re-direct so that the QR codes and outstanding links that point to `bonsai/quickstart` don't 404. 
- [ ] This PR deletes the `bonsai/overview.md` doc. We need to set up a re-direct so that QR codes and outstanding links that point to `bonsai/` don't 404. The redirect should point to `generating-proofs/remote-proving.md` 

<s>This PR also drops the slug on the `zkvm_overview` doc, which means links to `/zkvm/` will 404. If we want to drop this slug, we should set up a re-direct from `/zkvm/` to `/zkvm/zkvm_overview.md` to address this. </s> resolved by keeping the slug

flaub commented on website/api/generating-proofs/dev-mode.md:
```diff
@@ -1,3 +1,7 @@
+---
+slug: ./dev-mode
+---
+
```
This shouldn't be needed

flaub commented on website/api/use-cases.md:
Don't use `/api/next`, this makes it hard to snap a new release. Use relative paths, see: https://docusaurus.io/docs/versioning#link-docs-by-file-paths

flaub commented on website/api/use-cases.md:
```suggestion
1. [Write an application for the RISC Zero zkVM][zkvm-quickstart].
```

flaub commented on website/api/use-cases.md:
```suggestion
[zkvm-quickstart]: ./zkvm/quickstart.md
```

flaub commented on website/api/use-cases.md:
```suggestion
2. [Generate proofs for your zkVM application][bonsai-quickstart].
```

flaub commented on website/api/use-cases.md:
```suggestion
[bonsai-quickstart]: ./bonsai/quickstart.md
```

flaub commented on website/api/use-cases.md:
Maybe this sort of statement belongs on the marketing website, but it doesn't feel right for devloper docs.

nategraf commented on website/api/getting-started.md:
````diff
@@ -0,0 +1,61 @@
+```
+
+[zkvm-quickstart]: ./zkvm/quickstart.md
+[bonsai-quickstart]: ./bonsai/quickstart.md
````
`./bonsai/quickstart` appears to no longer be in the sidebar and is kind of stranded. it seems the remote proving page has all the information that used to be there, so linking there seems like the way to go, or simply to the "proving options" page. Either way, we should probably delete the Bonsai quickstart page.

nategraf reviewed (commented):
This looks quite good. Definitely a marked improvement over the current documentation structure.

Note that this will appear under "next" version, although all of it seems relevant to `0.20`. Once this sis essentially ready, it would be good to copy the changes there so folks land on the updated docs. (That or we simply release `0.21` very soon)

Cardosaum reviewed (commented):
The content is great and love the reorganization!

Made some comments but I think we could land this and iterate on a future PR if appropriate.

nategraf commented:
Additional note that once we are happy with the new structure, we should add redirects before merging this PR.

flaub commented on website/api/generating-proofs/local-proving.md:
```diff
@@ -0,0 +1,22 @@
+---
+slug: ./local-proving
+---
```
Can we drop all the slugs?

flaub commented on website/api/generating-proofs/remote-proving.md:
```diff
@@ -0,0 +1,38 @@
+[journal]: /terminology#journal
+[Bonsai Pay]: https://www.risczero.com/news/bonsai-pay
+[Bonsai Foundry Template]: https://github.com/risc0/bonsai-foundry-template/blob/main/README.md
+[cargo risczero]: https://crates.io/crates/cargo-risczero
```
Duplicate link

pdg744 commented on website/api/getting-started.md:
```diff
@@ -0,0 +1,61 @@
+
+[zkvm-quickstart]: ./zkvm/quickstart.md
+[bonsai-quickstart]: ./bonsai/quickstart.md
+[foundry-template]: https://github.com/risc0/bonsai-foundry-template
```
I'll defer to you and/or Angelo on that question.

pdg744 replied:
i've updated to point to bonsai-on-eth for now.

pdg744 commented on website/api/generating-proofs/remote-proving.md:
```diff
@@ -0,0 +1,38 @@
+- via the **[Bonsai SDK]** (for accessing Bonsai from Rust)
+- via the **[Bonsai REST API]** (for building your own client or accessing Bonsai directly)
+
+If you don't have an API key, you can [request access].
```
fixed

pdg744 commented on website/api/use-cases.md:
```diff
@@ -0,0 +1,82 @@
+emergent use cases which we are excited to enable. Key among these use cases
+are:
+
+- [zk coprocessors], which enable on-chain applications to reduce gas costs by
+  moving the complex part of their application logic off-chain
+- blockchain infrastructure, including [our work with Optimism][optimism]
```
good call-out!  

if you have a suggested revision i'm happy to include an improvement to this on this PR, otherwise let's save it for a future improvement -- I'm just copying/pasting this section from elsewhere on the site.

nategraf replied:
Ya, I dislike the way this section is written. I wrote some suggestions, although I'm still not totally very happy with it https://github.com/risc0/risc0/pull/1484

pdg744 replied:
awesome -- thanks victor!

pdg744 commented on website/api/use-cases.md:
```diff
@@ -0,0 +1,82 @@
+cryptography.
+
+These applications include:
+
```
good call!

pdg744 commented on website/api/zkvm/zkvm_overview.md:
```diff
@@ -7,36 +7,18 @@ slug: ./
 > WARNING: This software is still experimental, we do not recommend it for
 > production use.
```
feel free to open an issue for this; i'm not going to worry about it on this PR

pdg744 commented on website/api/zkvm/zkvm_overview.md:
```diff
@@ -68,19 +50,6 @@ For local proof generation, we recommend a machine with at least 16GB of RAM.
 Options for generating proofs using GPU acceleration and skipping proof
 generation are documented in the [feature flags].
```
i'm not sure what the best source of truth is for feature flags info

flaub replied:
I think currently it's the README.

pdg744 commented on website/api/getting-started.md:
```diff
@@ -0,0 +1,61 @@
+
+1. [Write an application for the RISC Zero zkVM][zkvm-quickstart].
+2. [Generate proofs for your zkVM application][bonsai-quickstart].
+3. [Integrate your proofs into on-chain applications][foundry-template].
```
interesting option to embed content from one page in another! i've revised to avoid the redundancy, but mb we want to take advantage of that embed option...

pdg744 commented on website/api/getting-started.md:
ya true. revised!

pdg744 commented:
thanks for the feedback, y'all! i think i've addressed all the feedback that it makes to address on this PR. 
if there's no further feedback, i'll connect with morgan to set up the appropriate re-directs and we can get this landed.

capossele commented:
This is a great improvement! I have few minor feedbacks that do not necessarily need to be considered within this PR.
- I feel there is a gap between the getting started and the actual information the user needs to get started without getting distracted by secondary details. I'm mostly referring to the **What is a zkVM application** section. In there we have some *101s* that I personally find not that easy to follow unless knowing already a basic end to end flow. 
- In section **Writing your zkVM Application** we have a bunch of different type of resources; I'm not convinced about having the zkVM spec in this section. Also I feel the *rust resources* should be moved after the tutorial.

nategraf commented:
> thanks for the feedback, y'all! i think i've addressed all the feedback that it makes to address on this PR. if there's no further feedback, i'll connect with morgan to set up the appropriate re-directs and we can get this landed.
> 
> feel free to add more comments if there are other things that we should improve here.

I'm happy with these changes, and think this is a marked improvement. I'd be happy to see further content work happen in follow-up PRs.

pdg744 commented:
OK the only remaining todo here is to make sure we're not killing links from devrel/marketing assets. I think we can manage the redirects via the docusaurus.config.js file, but I struggled to figure it out when I tried earlier. If someone's available to pair with me for 30 minutes, I think we can land this! 

(Here's a list of links we need to be careful about: https://docs.google.com/spreadsheets/d/1U3CAIzdsuX4eQQIfof9wkCXt0LYRjF8hp9KeK8uPVyY/edit#gid=0)

flaub commented on website/api/zkvm/zkvm_overview.md:
```diff
@@ -1,42 +1,20 @@
----
-slug: ./
----
-
```
I think we need to keep this one

flaub commented on website/api/introduction.md:
```diff
@@ -1,30 +1,8 @@
----
-slug: /
----
-
```
I think we need to keep this one too
